projects/extension/src/background/ConnectionManager.ts

```diff
@@ -252,11 +278,24 @@ export class ConnectionManager
       ? this.#networks.filter((n) => n.tabId === tabId)
       : this.#networks
 
-    const addedChain = await this.#client.addChain({
-      chainSpec,
-      jsonRpcCallback,
-      potentialRelayChains: potentialNetworks.map((r) => r.chain),
-    })
+    const databaseContent = await this.getObjectFromLocalStorage(
+      (name as string).toLowerCase(),
+    )
+
+    const options = databaseContent
+      ? {
+          chainSpec,
+          jsonRpcCallback,
+          databaseContent,
+          potentialRelayChains: potentialNetworks.map((r) => r.chain),
+        }
+      : {
+          chainSpec,
+          jsonRpcCallback,
+          potentialRelayChains: potentialNetworks.map((r) => r.chain),
+        }
```

Comment:
This ternary is bogus

```suggestion
    const options: AddChainOptions =  {
        chainSpec,
        jsonRpcCallback,
        databaseContent,
        potentialRelayChains: potentialNetworks.map((r) => r.chain),
    }
```